src/promql/src/planner.rs

```diff
@@ -0,0 +1,627 @@
+                let name = name.as_ref().unwrap().clone();
+                self.setup_context(&name)?;
+                let normalize =
+                    self.selector_to_series_normalize_plan(&name, *offset, label_matchers.clone())?;
+                let manipulate = InstantManipulate::new(
+                    self.ctx.start,
+                    self.ctx.end,
+                    self.ctx.lookback_delta,
+                    self.ctx.interval,
+                    self.ctx
+                        .time_index_column
+                        .clone()
+                        .with_context(|| TimeIndexNotFoundSnafu { table: name })?,
+                    normalize,
+                );
+                LogicalPlan::Extension(Extension {
+                    node: Arc::new(manipulate),
+                })
+            }
+            PromExpr::MatrixSelector { .. } => UnsupportedExprSnafu {
+                name: "Prom Matrix Selector",
+            }
+            .fail()?,
+            PromExpr::Call { func, args } => {
+                let args = self.create_function_args(args)?;
+                let input =
+                    self.prom_expr_to_plan(args.input.with_context(|| ExpectExprSnafu {
+                        expr: prom_expr.clone(),
+                    })?)?;
+                let mut func_exprs = self.create_function_expr(func, args.literals)?;
+                func_exprs.insert(0, self.create_time_index_column_expr()?);
+                LogicalPlanBuilder::from(input)
+                    .project(func_exprs)
+                    .context(DataFusionPlanningSnafu)?
+                    .build()
+                    .context(DataFusionPlanningSnafu)?
+            }
+        };
+        Ok(res)
+    }
+
+    fn selector_to_series_normalize_plan(
+        &self,
+        table_name: &str,
+        offset: Option<Duration>,
+        label_matchers: Matchers,
+    ) -> Result<LogicalPlan> {
+        // TODO(ruihang): add time range filter
+        let filter = self.matchers_to_expr(label_matchers)?;
+        let table_scan = self.create_table_scan_plan(table_name, filter)?;
+        let offset = offset.unwrap_or_default();
+
+        let series_normalize = SeriesNormalize::new(
+            offset,
+            self.ctx
+                .time_index_column
+                .clone()
+                .with_context(|| TimeIndexNotFoundSnafu { table: table_name })?,
+            table_scan,
+        );
+        let logical_plan = LogicalPlan::Extension(Extension {
+            node: Arc::new(series_normalize),
+        });
+        Ok(logical_plan)
+    }
+
+    // TODO(ruihang): ignore `MetricNameLabel` (`__name__`) matcher
+    fn matchers_to_expr(&self, label_matchers: Matchers) -> Result<Vec<DfExpr>> {
+        let mut exprs = Vec::with_capacity(label_matchers.matchers.len());
+        for matcher in label_matchers.matchers {
+            let col = DfExpr::Column(Column::new(None::<String>, matcher.name));
```

Comment:
Looks like you could use [Column::from_name](https://docs.rs/datafusion/latest/datafusion/prelude/struct.Column.html#method.from_name) to replace `Column::new()` in the codes.